### PR TITLE
adds signIn and signUp pages

### PR DESCRIPTION
**Pull request by BulatCC**

### CODE REVIEW

File: client/src/api/requests/signUp.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { post } from 'src/api/api'
+import { SignUpFormServerData } from 'src/components/SignUpForm/SignUpForm'
```
Comment (Nurai1) [inline client/src/api/requests/signUp.ts:2]: why do u write types by yourself, they generate automatically

File: client/src/components/SignInForm/SignInForm.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	const onChange = (value?: string, name?: string) => {
+		setFormData({
+			...formData,
+			[name as string]: value as string
+		})
+	}
+
+	useEffect(() => {
+		setValidateErrors(validator(formData, validatorConfigSingIn))
+	}, [formData])
```
Comment (Nurai1) [inline client/src/components/SignInForm/SignInForm.tsx:39]: why so many effects? let's do validation on submit, delete validation error on focus

File: client/src/components/SignInForm/SignInForm.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+	const fillEmptyInputs = () => {
```
Comment (Nurai1) [inline client/src/components/SignInForm/SignInForm.tsx:41]: why? weird function

File: client/src/helpers/formValidators.ts [hunk 4]
```diff
@@ -12,3 +12,9 @@ export const emailStringValidator = nonEmptyStringValidator.email()
 
 const URL_REGEX = /^(?:http(s)?:\/\/)?[\w.-]+(?:\.[\w.-]+)+[\w\-._~:/?#[\]@!$&'()*+,;=.]+$/m
 export const invalidUrlStringValidator = z.custom<string>(val => URL_REGEX.test(val as string), 'Invalid url')
+
+// eslint-disable-next-line
```
Comment (Nurai1) [inline client/src/helpers/formValidators.ts:16]: why?

File: client/src/hooks/useRedirectAuthorizedUser.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { useEffect } from 'react'
+import { useAppSelector } from 'src/hooks/redux'
+import { AuthorizationStatus } from 'src/constants/authorizationStatus'
+import { useNavigate } from 'react-router-dom'
+
+export const useRedirectAuthorizedUser = (path: string) => {
+	const authStatus = useAppSelector(state => state.user.authorizationStatus)
+	const navigate = useNavigate()
+
+	useEffect(() => {
```
Comment (Nurai1) [inline client/src/hooks/useRedirectAuthorizedUser.ts:10]: where are dependencies?

File: client/src/routes/CompetitionsPage.tsx [hunk 6]
```diff
@@ -21,14 +21,16 @@ export const CompetitionsPage = (): ReactElement => {
...
 
 	const allCompetitions = useAppSelector(selectCompetitions)
 	const competitionsActive = useAppSelector(activeCompetitionsSelector)
-	const currentUserCompetitions = useAppSelector(currentUserCompetitionsSelector)
+	const currentUserCompetitions = useAppSelector(currentUserCompetitionsSelector(authorizedUserId))
 	const expiredCompetitions = useAppSelector(expiredCompetitionsSelector)
 
 	useEffect(() => {
 		if (!allCompetitions.length) dispatch(fetchCompetitions())
+		// eslint-disable-next-line
```
Comment (Nurai1) [inline client/src/routes/CompetitionsPage.tsx:33]: again no specific error

File: client/src/api/requests/signUp.ts (hunk 1 above)
Comment (BulatCC) [reply client/src/api/requests/signUp.ts:2]: All fields are optional, for live validation. Now it's not needed

File: client/src/helpers/formValidators.ts (hunk 4 above)
Comment (BulatCC) [reply client/src/helpers/formValidators.ts:16]: linter says regex has useless excape characters, but when i delete them, regex doesn't work

**Pull request merged**